addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -777,6 +760,34 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
             });
         }
 
+        private String constructCoverArtUrl(String mac, boolean coverart, String coverid, String artwork_url) {
+            String hostAndPort = "http://" + host + ":" + webport;
+
+            // Default to using the convenience artwork URL (should be rare)
+            String url = hostAndPort + "/music/current/cover.jpg?player=" + encode(mac);
+
+            // If additional artwork info provided, use that instead
+            if (coverart) {
+                if (coverid != null) {
+                    // Typically is used to access cover art of local music files
+                    url = hostAndPort + "/music/" + coverid + "/cover.jpg";
+                }
+            } else if (artwork_url != null) {
+                if (artwork_url.startsWith("http") || artwork_url.startsWith("https")) {
```

Comment:
I assume that everything that starts with 'https' also starts with 'http', wdyt?

Reply:
Of course!